src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -0,0 +1,412 @@
+            pmwNewVendorAttributesConformToKfsLengthsOrFormats(stgNewVendorRequestDetailToProcess, errorMessages) &&
+            allPmwNewVendorIsoCountriesMapToSingleFipsCountry(stgNewVendorRequestDetailToProcess, errorMessages) &&
+            pmwNewVendorIdentifierDoesNotExistInKfsStagingTable(stgNewVendorRequestDetailToProcess, errorMessages)){
+            return true;
+        }
+        else {
+            if (!errorMessages.isEmpty()){
+                reportData.getPendingPaymentWorksVendorsThatCouldNotBeProcessed().incrementRecordCount();
+                reportData.addPmwVendorsThatCouldNotBeProcessed(new PaymentWorksBatchReportRawDataItem(stgNewVendorRequestDetailToProcess.toString(), errorMessages));
+            }
+            return false;
+        }
+    }
+    
+    private boolean pmwDtosCouldConvertCustomAttributesToPmwJavaClassAttributes(PaymentWorksVendor stgNewVendorRequestDetailToProcess) {
+        if (stgNewVendorRequestDetailToProcess.isCustomFieldConversionErrors()) {
+            return false;
+        }
+        else{
+            return true;
+        }
+    }
+    
+    private boolean pmwNewVendorAttributesConformToKfsLengthsOrFormats(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
+        boolean allValidationPassed = true;
+        if (StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName()) &
+            StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName()))  {
+            int lastDelimiterFirstCombinedLength = stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName().length() + 
+                                                   VendorConstants.NAME_DELIM.length() + 
+                                                   stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName().length();
+            
+            if (lastDelimiterFirstCombinedLength > VendorConstants.MAX_VENDOR_NAME_LENGTH) {
+              errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.NEW_VENDOR_REQUEST_COMBINED_LEGAL_FIRST_LAST_NAME_TOO_LONG_FOR_KFS);
+              allValidationPassed = false;
+              LOG.info("pmwNewVendorAttributesConformToKfsAttributesMaxLengths: PMW Legal Last Name length is " + stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName().length() +
+                       " PMW Legal First Name length is " + stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName().length() +
+                       " which when combined with delimiters must be less that the KFS legal name max length of " + (VendorConstants.MAX_VENDOR_NAME_LENGTH - VendorConstants.NAME_DELIM.length()));
+            }
+        }
+        else {
+            if (StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalName())) {
```

Comment:
could this be extracted in a separate method?

Reply:
Changed as requested.